jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamSchemasResultSet.java

```diff
@@ -15,22 +15,100 @@
 package software.amazon.timestream.jdbc;
 
 import com.amazonaws.services.timestreamquery.model.ColumnInfo;
+import com.amazonaws.services.timestreamquery.model.Datum;
+import com.amazonaws.services.timestreamquery.model.Row;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Result set to return an empty list of schemas in Timestream.
  */
-public class TimestreamSchemasResultSet extends TimestreamEmptyBaseResultSet {
+public class TimestreamSchemasResultSet extends TimestreamBaseResultSet {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TimestreamDatabasesResultSet.class);
+  private static final Datum NULL_DATUM = new Datum().withNullValue(Boolean.TRUE);
   private static final List<ColumnInfo> COLUMNS = ImmutableList.of(
     TimestreamDataType.createColumnInfo(TimestreamDataType.VARCHAR, "TABLE_SCHEM"),
     TimestreamDataType.createColumnInfo(TimestreamDataType.VARCHAR, "TABLE_CATALOG"));
 
+  private boolean isAfterLast = false;
+
   /**
    * Constructor.
+   *
+   * @param connection the parent connection of the result set.
+   * @param schemaPattern the SchemaPattern. null if not provided
+   * @throws SQLException if a database access error occurs.
    */
-  public TimestreamSchemasResultSet() {
-    super(COLUMNS);
+  TimestreamSchemasResultSet(TimestreamConnection connection, String schemaPattern) throws SQLException {
+    super(null, 20);
+    this.rsMeta = createColumnMetadata(COLUMNS);
+
+    populateCurrentRows(connection, schemaPattern);
+  }
+
+  @Override
+  public boolean isAfterLast() throws SQLException {
+    verifyOpen();
+    return isAfterLast;
+  }
+
+  @Override
+  public boolean isLast() throws SQLException {
+    verifyOpen();
+    LOGGER.debug("Checking whether the last row of this TimestreamSchemasResultSet has been reached.");
```

Comment:
Is there a reference documentation on how does calling application (like this sample: https://github.com/awslabs/amazon-timestream-tools/tree/mainline/integrations/jdbc ) can enable printing debug logs to console?

Reply:
This driver is using slf4j facade for logging. This means the application (in this case the provided sample) should provide the implementation for logging. There are multiple slf4j implementations, one of the common is logback. Example of how to configure it is available here - https://mkyong.com/logging/slf4j-logback-tutorial/.

In order to enable debug logs [this application](https://github.com/awslabs/amazon-timestream-tools/tree/mainline/integrations/jdbc) should be adjusted in the following way - 
1. Add Dependency with slf4j implementation, e.g.
```<dependency>
            <groupId>ch.qos.logback</groupId>
            <artifactId>logback-classic</artifactId>
            <version>1.4.5</version>
        </dependency>
```

2. Add logback.xml configuration file to `resources` folder:
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
        <layout class="ch.qos.logback.classic.PatternLayout">
            <Pattern>
                %d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n
            </Pattern>
        </layout>
    </appender>
    <root level="debug">
        <appender-ref ref="CONSOLE"/>
    </root>
</configuration>
```

This is one of the possible options to configure slf4j. Application may choose to use other logging implementations, like log4j, jul, etc.